music_assistant/providers/bbc_sounds/__init__.py

```diff
@@ -0,0 +1,860 @@
+) -> tuple[ConfigEntry, ...]:
+    """
+    Return Config entries to setup this provider.
+
+    instance_id: id of an existing provider instance (None if new instance setup).
+    action: [optional] action key called from config entries UI.
+    values: the (intermediate) raw values for config entries sent with the action.
+    """
+    # ruff: noqa: ARG001
+
+    return (
+        ConfigEntry(
+            key=_Constants.CONF_INTRO,
+            type=ConfigEntryType.LABEL,
+            label="A BBC Sounds account is optional, but some UK-only content may not work without"
+            " it",
+        ),
+        ConfigEntry(
+            key=CONF_USERNAME,
+            type=ConfigEntryType.STRING,
+            label="Email or username",
+            required=False,
+        ),
+        ConfigEntry(
+            key=CONF_PASSWORD,
+            type=ConfigEntryType.SECURE_STRING,
+            label="Password",
+            required=False,
+        ),
+        ConfigEntry(
+            key=_Constants.CONF_SHOW_LOCAL,
+            category="advanced",
+            type=ConfigEntryType.BOOLEAN,
+            label="Show local radio stations?",
+            default_value=False,
+        ),
+        ConfigEntry(
+            key=_Constants.CONF_NOW_PLAYING,
+            category="advanced",
+            type=ConfigEntryType.BOOLEAN,
+            label="Show 'now playing' details?",
+            description=(
+                "Show details of the currently playing track instead of the station details"
+            ),
+            default_value=True,
+        ),
+        ConfigEntry(
+            key=_Constants.CONF_UPDATE_INTERVAL,
+            category="advanced",
+            type=ConfigEntryType.INTEGER,
+            label="Player update interval (seconds)",
+            description="How often to check for now playing updates",
+            default_value=5,
+            range=(1, 60),
+            depends_on="now_playing",
+            depends_on_value=True,
+        ),
+        ConfigEntry(
+            key=_Constants.CONF_RECOMMENDATIONS,
+            category="advanced",
+            label="Show recommendations?",
+            description="BBC Sounds has several recommendation categories, configure if "
+            "and where these are shown",
+            type=ConfigEntryType.STRING,
+            options=[
+                ConfigValueOption(
+                    "Show recommendations on the home page",
+                    _Constants.CONF_RECOMMENDATIONS_HOMEPAGE,
+                ),
+                ConfigValueOption(
+                    "Show recommendations in folders on the browse page",
+                    _Constants.CONF_RECOMMENDATIONS_BROWSE,
+                ),
+                ConfigValueOption(
+                    "Disable recommendations", _Constants.CONF_RECOMMENDATIONS_DISABLE
+                ),
+            ],
+            default_value=_Constants.CONF_RECOMMENDATIONS_HOMEPAGE,
+            required=True,
+        ),
```

Comment:
Why make this configurable ? Just return recommendations and if the user doesnt want them, they can hide the rows on the homescreen